app/_src/gateway/production/debug-request.md

```diff
@@ -59,6 +59,14 @@ If the `X-Kong-Request-Debug-Log` header is set to true, timing information will
 
 The `X-Kong-Request-Debug-Token` is a token for authenticating the client and making the debug request to prevent abuse. Debug requests originating from loopback addresses don't require this header.
 
+{% if_version gte:3.5.x %}
+### X-Kong-Request-Id header
+
+The `X-Kong-Request-Id` header contains a unique identifier for each client request. This is enabled by default both upstream and downstream. This unique ID helps in matching specific requests to their corresponding error logs, which is useful for debugging. If {{site.base_gateway}} returns an error, the request ID will also be included in the response body generated by {{site.base_gateway}}. In addition, any {{site.base_gateway}} error log generated will contain the same request ID, with the format: `request_id: xxx`.
+
+The log line produced by the debug header as well as the debug response header contains the request ID. You can use this to search for the debug header output using a log viewer UI. This is especially useful when the debug output is too long to fit in the response header.
```

Comment:
Can we add an example log line here?

Reply:
@lena-larionova There was a log example that included request id later on in the page in a different section, so I added a link to that instead of using the same/another example. Does that work?

Reply:
That's great!